orchagent/vnetorch.cpp

```diff
@@ -960,62 +1098,110 @@ bool VNetRouteOrch::doRouteTask<VNetVrfObject>(const string& vnet, IpPrefix& ipP
             {
                 SWSS_LOG_ERROR("Route add/update failed for %s, vr_id '0x%" PRIx64, ipPrefix.to_string().c_str(), vr_id);
                 /* Clean up the newly created next hop group entry */
-                if (nexthops.getSize() > 1)
+                if (active_nhg.getSize() > 1)
                 {
-                    removeNextHopGroup(vnet, nexthops, vrf_obj);
+                    removeNextHopGroup(vnet, active_nhg, vrf_obj);
                 }
                 return false;
             }
         }
-
-        if (it_route != syncd_tunnel_routes_[vnet].end() && it_route->second != nexthops)
+        bool route_updated = false;
+        bool priority_route_updated = false;
```

Comment:
What does priority_route_updated stand for? Please add a comment for clarity

Reply:
the priority route is the handling of this scenario where  we can switch between primary and secondary. The case is where secondary nexthop is not empty and monitoring is custom.